src/main/java/com/parasoft/findings/utils/common/util/ZipFileUtil.java

```diff
@@ -0,0 +1,78 @@
+package com.parasoft.findings.utils.common.util;
+
+import com.parasoft.findings.utils.common.IStringConstants;
+
+import java.io.*;
+import java.util.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public final class ZipFileUtil {
+    private static String ruleDocZipFilePath;
+
+    /**
+     * Private constructor to prevent instantiation.
+     */
+    private ZipFileUtil() {
+    }
+
+    /**
+     * Read the contents of a rule document file from a provided zip file
+     *
+     * @param filePathInZip  the file path in the zip file
+     * @param sEncoding      encoding file, example: UTF-8
+     * @return file contents
+     * @throws IOException
+     */
+    public static String readRuleDocFileInZip(String filePathInZip, String sEncoding) throws IOException {
+        try (ZipFile zipFile = new ZipFile(ruleDocZipFilePath)){
+            ZipEntry entry = zipFile.getEntry(filePathInZip);
+            if (entry == null) {
+                return IStringConstants.EMPTY;
+            }
+
+            try (InputStream inputStream = zipFile.getInputStream(entry);
+                 InputStreamReader streamReader = new InputStreamReader(inputStream, sEncoding);
+                 BufferedReader reader = new BufferedReader(streamReader)) {
+                return FileUtil.readFile(reader);
+            }
+        }
+    }
+
+    /**
+     * Get the rule doc file location in the specified zip file
+     *
+     * @param zipFilePath the absolute path of the zip file
+     * @param ruleId      rule id
+     * @return the rule file path in the specified zip file
+     */
+    public static String getRuleDocFileLocationInZip(String zipFilePath, String ruleId) {
+        List<String> baseDirs = new ArrayList<>(Arrays.asList("doc", "docs"));
+        Map<String, String> languageSubdirMap = new HashMap<>();
+        languageSubdirMap.put(Locale.CHINESE.getLanguage(), "zh_CN/");
+        languageSubdirMap.put(Locale.JAPANESE.getLanguage(), "ja/");
+
+        try (ZipFile zipFile = new ZipFile(zipFilePath)) {
+            // Find the rule doc directory (doc or docs) in the zip
+            String ruleDocDir = baseDirs.stream()
```

Comment:
What if `ruleDocDir == null`?

Reply:
Additional note: While using AI to learn about `ZipEntry`, I found that the way a zip file is packaged can affect whether we can directly match the name to find `doc` or `docs`. I'm not sure if we need to take this into consideration (e.g. later they changed their way to package zip), but if we do, the logic might need to account for two scenarios: either directly finding an entry named `doc` or `docs`, or finding entries whose names start with `doc/` or `docs/`.